services/wallet/api.go

```diff
@@ -313,6 +313,12 @@ func (api *API) GetCryptoOnRamps(ctx context.Context) ([]onramp.CryptoOnRamp, er
    Collectibles API Start
 */
 
+func (api *API) FetchBalancesByOwnerAndContractAddressFromCache(ctx context.Context, chainID wcommon.ChainID, ownerAddress common.Address, contractAddresses []common.Address) (thirdparty.TokenBalancesPerContractAddress, error) {
```

Comment:
Do we need to call this from the client?

Reply:
Hum yes..., it's somehow `polymorphically` called from the `GetOwnedERC721Tokens` in order to retrieve the tokens by the member.
Please clarify if I didn't understand the question.

Reply:
Since it's in api.go, it means it can be called by clients (Nim). Is that necessary?

Reply:
> Since it's in api.go, it means it can be called by clients (Nim). Is that necessary?

Fair point, after second thought, we might just want to hide that from being called by clients indeed. I'll submit another revision